### PR TITLE
The CL can now recruit PFCs

### PR DESCRIPTION
**Pull request by Zonespace27**
## About The Pull Request
**Code bounty, concept, and armor sprites by Boperator**

Original IdeasGuy thread: https://discord.com/channels/150315577943130112/1035410369323212820

Dev approval (somewhat)

image

The CL can now recruit up to 4 PFCs (prior to first drop) to act as ship-side PMC recruits who aren't allowed to deploy, instead being effectively a ship-side RP role.

## Exactly what it is

There's two new forms (which would be available through the CL's vendor if [field] worked when placed via code), the "immediate PMC transfer" and "future PMC transfer" forms.

- Future:
Once the signer's employment with the USMC has ended, they will become employed with the W-Y PMC corps.

- Immediate:
Effective immediately (the CL should make sure things are handled smoothly with getting the PFC discharged smoothly prior), the signer is a W-Y PMC recruit. They're able to use, with the help of the CL, a PMC recruiting console in their office, which following a hand scan and the confirmation that the appropriate paperwork has been filed, transfers the PFC. This means their faction is changed, ID role changed to "Weyland-Yutani PMC Recruit", and they are removed from the ship-side manifest. In addition, the CL can help them choose from a few different sets of (not-quite) PMC gear (light armor/medium armor/heavy armor) + (helmet/cap/beret), which is for the most part reskinned marine equipment, with no upsides to the latter. The CL gets a monitoring console in their office for all PMC recruits.

## Papercode

<details>
<summary>Immediate PMC Transfer</summary>

```
{"\[center\]\[wy\]
		\[b\]\[large\]Building Better Worlds.\[/large\]\[/b\]\[/br\]
		\[small\]Copyright Weyland-Yutani Corporation, 2182. All Rights Reserved.\[/center\]\[/small\]
		\[hr\]\[hr\]
		\[center\]\[b\]\[u\]Form 8303
		Weyland-Yutani Private Military Enlistment Transfer, "Whiteguard Solutions"\[/u\]\[/b\]\[br\]
		\[b\]Date:\[/b\] \[date\]\[/center\]
		\[hr\]\[hr\]
		\[br\]
		\[list\]Location of transfer: \[b\]USS Almayer\[/b\]
		\[br\]
		\[list\]\[i\]I, \[field\], willingly enter into this contract and agree to all written terms and conditions. I understand that once I have signed and adhered to the conditions of this contract, I will be binded to such as is written until the term agreed upon expires or is deemed void by the appropriate personnel.\[/i\]\[/list\]\[/list\]
		\[hr\]\[hr\]

		\[list\]\[*\]The signatory herein agrees to act as a Private Military Contractor for the company of Weyland-Yutani for at least four (4) years before they are eligible to opt out of the position and pursue different employment.\[br\]
		\[*\]The signatory understands that they will not be eligible for any and all company-provided health and other benefits until they have served under this contract for at least one (1) year.\[br\]
		\[*\]The signatory agrees and comprehends that this contract may be terminated at any given time, without advance notice, by the corporation if the signed is found to be lacking or otherwise failing in their duties and overall performance as a military contractor.\[br\]
		\[*\]The signatory has provided evidence and has submitted to a sufficient background check to verify their previous work history and combat experience; as well as provided previously-recorded medical history pretaining to any performance-affecting injuries they may have received while assuming their duties while under previous employment.\[br\]
		\[*\]The signatory agrees that, while they are binded under the terms and conditions of this contract, they will \[b\]not\[/b\] work or otherwise perform acts of duty for other companies or military organizations, and will only employ their services as required by Weyland-Yutani executive staff or their assigned squad commander (if they have one). The signed agrees that they will not deploy to respond to mission directives not provided by Weyland-Yutani corporation, and as such, will not work to actively assist members of the United States Colonial Marines or members of any other outstanding factions, including but not limited to: United Progressive Peoples, Colonial Liberation Front; unless otherwise permitted by Weyland-Yutani executive staff of rank Division Manager or above.\[br\]
		\[*\]Company-provided equipment is a privilege that can be revoked at any time.\[br\]
		\[*\]The signatory agrees that written documentation and word of mouth with relation to their work performed while contracted under Weyland-Yutani will not be disclosed or otherwise revealed to those not affiliated with the corporation. A breach of privacy will result in immediate contract termination without advance notice or pay.\[/list\]\[br\]\[hr\]\[hr\]\[br\]
		\[list\]Signature: | \[field\]\[br\]
		Corporate Liaison Signature: |\[field\]\[/list\]\[hr\]\[hr\]
		\[center\]\[small\]Weyland-Yutani Corporation does not take financial or legal responsibility for any loss of life or limb that may occur as a result of employment under this contract.\[br\]
		\[b\]Weyland-Yutani Corporate Employment Records Forms 268-C, section 1.3.\[/b\]\[/small\]"}
```

</details>

<details>
<summary>Future PMC Transfer</summary>

```
{"\[center\]\[wy\]
		\[b\]\[large\]Building Better Worlds.\[/large\]\[/b\]
		\[small\]Copyright Weyland-Yutani Corporation, 2182. All Rights Reserved.\[/center\]\[/small\]
		\[hr\]\[hr\]
		\[center\]\[b\]\[u\]Form 8302
		Weyland-Yutani Private Military Enlistment Transfer, "Whiteguard Solutions"\[/u\]\[/b\]\[br\]
		\[b\]Date:\[/b\] \[date\]\[/center\]
		\[hr\]\[hr\]
		\[list\]\[b\]Current Facility:\[/b\] USS Almayer\[br\]
		\[list\]\[i\]I, \[field\], willingly enter into this contract and agree to all written terms and conditions. I understand that once signed, I will be bound to such as is written until the term agreed upon expires or is deemed void by the appropriate Weyland-Yutani personnel. I understand that this contract is to take effect once my term with my present employer has concluded and I am free of any possible conflict of interest.\[/i\]\[/list\]\[/list\]
		\[hr\]\[hr\]\[list\]\[*\]The signatory herein agrees to act as a Private Military Contractor for the company of Weyland-Yutani for at least four (4) years before they are eligible to opt out of the position and pursue different employment.\[br\]
		\[*\]The signatory understands that they will not be eligible for any and all company-provided health and other benefits until they have served under this contract for at least one (1) year.\[br\]
		\[*\]The signatory agrees and comprehends that this contract may be terminated at any given time, without advance notice, by the corporation if the signed is found to be lacking or otherwise failing in their duties and overall performance as a military contractor.\[br\]
		\[*\]The signatory has provided evidence and has submitted to a sufficient background check to verify their previous work history and combat experience; as well as provided previously-recorded medical history pertaining to any performance-affecting injuries they may have received while assuming their duties while under previous employment.\[br\]
		\[*\]Company-provided equipment is a privilege that can be revoked at any time.\[br\]
		\[*\]The signatory agrees that written documentation and word of mouth with relation to their work performed while contracted under Weyland-Yutani will not be disclosed or otherwise revealed to those not affiliated with the corporation. A breach of privacy will result in immediate contract termination without advance notice or pay.\[/list\]\[hr\]\[hr\]\[list\]\[b\]Signature:\[/b\] \[field\]\[br\]
		\[b\]Corporate Liaison Signature:\[/b\] \[field\]\[/list\]\[hr\]\[hr\]\[center\]\[small\]Weyland-Yutani Corporation does not take financial or legal responsibility for any loss of life or limb that may occur as a result of employment under this contract.\[br\]
		\[b\]Weyland-Yutani Corporate Employment Records Forms 268-C, section 1.3.\[/b\]\[/small\]"}
```

</details>

## Images

All three armors and headgears

image

image

image

image

CL's recruit crew monitor

image

The W-Y IFF granter/remover

image

## Why It's Good For The Game
It opens up new avenues and tools for RP, especially for the usually-ignored PFCs, and also gives some background/reasoning on how W-Y gets their PFCs in the first place

## Changelog

:cl:
add: The CL can now recruit up to 4 PFCs into being a recruit PMC, an RP role.
/:cl:

### CODE REVIEW

File: tgui/packages/tgui/interfaces/PmcTransfer.js
Comment (harryob) [inline tgui/packages/tgui/interfaces/PmcTransfer.js:21]:
```suggestion
  if (verification) {
```

File: tgui/packages/tgui/interfaces/PmcTransfer.js
Comment (harryob) [inline tgui/packages/tgui/interfaces/PmcTransfer.js:103]:
```suggestion
      {(!!(possible_verifications <= 0) && <NoRecruitsDimmer />)}
```

no empty fragments

File: tgui/packages/tgui/interfaces/PmcTransfer.js
Comment (harryob) [inline tgui/packages/tgui/interfaces/PmcTransfer.js:20]:
```suggestion
  let primary;
```

File: tgui/packages/tgui/interfaces/PmcTransfer.js
Comment (harryob) [inline tgui/packages/tgui/interfaces/PmcTransfer.js:134]:
```suggestion
      {(is_loading && <LoadingScreenDimmer />)}
```

File: tgui/packages/tgui/interfaces/PmcTransfer.js
Comment (harryob) [inline tgui/packages/tgui/interfaces/PmcTransfer.js:135]:
```suggestion
      {(!!(possible_verifications <= 0) && <NoRecruitsDimmer />)}
```

File: tgui/packages/tgui/interfaces/PmcTransfer.js
Comment (harryob) [inline tgui/packages/tgui/interfaces/PmcTransfer.js:55]:
```suggestion
      {(!!(possible_verifications <= 0) && <NoRecruitsDimmer />)}
```

File: tgui/packages/tgui/interfaces/PmcTransfer.js
Comment (harryob) [inline tgui/packages/tgui/interfaces/PmcTransfer.js:125]:
```suggestion
      268-C and any paperwork the recruit&apos;s former employer may require.
```
or escape the apostrophe some neater way

**Review by harryob — CHANGES_REQUESTED** [review]

File: tgui/packages/tgui/interfaces/PmcTransfer.js
Comment (Watermelon914) [reply tgui/packages/tgui/interfaces/PmcTransfer.js:103]: lol that statement should just be possible_verifications > 0

Comment (harryob) [reply tgui/packages/tgui/interfaces/PmcTransfer.js:103]:
```suggestion
      {possible_verifications <= 0 && <NoRecruitsDimmer />}
```
walter's right, there's no need to booleanise here - you can do the same to the other instances

**Comment by Zonespace27** [comment]
> for the forms - did you try inserting a span class for the text input?
> 
> ```
> <span class="paper_field"></span>
> ```

yep, didn't work

File: tgui/packages/tgui/interfaces/PmcTransfer.js
Comment (harryob) [inline tgui/packages/tgui/interfaces/PmcTransfer.js:55]:
```suggestion
      {possible_verifications <= 0 && <NoRecruitsDimmer />)}
```

**Review by harryob — CHANGES_REQUESTED** [review]

File: tgui/packages/tgui/interfaces/PmcTransfer.js
Comment (harryob) [inline tgui/packages/tgui/interfaces/PmcTransfer.js:135]:
```suggestion
      {possible_verifications <= 0 && <NoRecruitsDimmer />}
```
oops - i messed up your parenthesis

File: tgui/packages/tgui/interfaces/PmcTransfer.js
Comment (harryob) [inline tgui/packages/tgui/interfaces/PmcTransfer.js:55]:
```suggestion
      {possible_verifications <= 0 && <NoRecruitsDimmer />}
```

**Review by harryob — CHANGES_REQUESTED** [review]

File: code/datums/datacore.dm
Comment (Zonespace27) [inline code/datums/datacore.dm:329]:
```suggestion
/datum/datacore/proc/manifest_remove(ref, name)
```

**Comment by Zonespace27** [comment]
Changes made after Boperator talked with GAs:

- Survivors can now be recruited
- Recruitment text is more clear that PFCs being recruited without getting a CO discharge is illegal from a ML standpoint
- People can be recruited any time now (as opposed to first drop)
- CL has a device that adds or removes W-Y IFF from an inserted ID, requires WY access to use
- The provided coif now hides your face

File: code/game/machinery/computer/pmc_transfer.dm
Comment (harryob) [inline code/game/machinery/computer/pmc_transfer.dm:3]: USCM

File: code/game/machinery/computer/pmc_transfer.dm
Comment (harryob) [inline code/game/machinery/computer/pmc_transfer.dm:27]: explicit ui_status proc too

File: code/game/machinery/computer/pmc_transfer.dm
Comment (harryob) [inline code/game/machinery/computer/pmc_transfer.dm:71]: SPAN_BOLD

File: code/game/machinery/computer/pmc_transfer.dm
Comment (harryob) [inline code/game/machinery/computer/pmc_transfer.dm:133]: SPAN_BOLD for all of these

File: code/game/machinery/computer/pmc_transfer.dm
Comment (harryob) [inline code/game/machinery/computer/pmc_transfer.dm:149]: SPAN_BOLD in this proc too

**Review by harryob — CHANGES_REQUESTED** [review]

File: code/game/machinery/computer/pmc_transfer.dm
Comment (Zonespace27) [reply code/game/machinery/computer/pmc_transfer.dm]: Performed all changes

File: code/game/machinery/computer/pmc_transfer.dm
Comment (harryob) [inline code/game/machinery/computer/pmc_transfer.dm:95]: O -> held_object or summat

File: code/game/machinery/computer/pmc_transfer.dm
Comment (harryob) [inline code/game/machinery/computer/pmc_transfer.dm:102]: as above

File: code/game/objects/items/devices/iff_granter.dm
Comment (harryob) [inline code/game/objects/items/devices/iff_granter.dm:22]: W -> attempted_item, something like that ig

File: code/game/objects/items/devices/iff_granter.dm
Comment (harryob) [inline code/game/objects/items/devices/iff_granter.dm:23]: as above

File: code/game/objects/items/devices/iff_granter.dm
Comment (harryob) [inline code/game/objects/items/devices/iff_granter.dm:82]:
```suggestion
```

Comment (harryob) [reply code/game/objects/items/devices/iff_granter.dm:82]: this is meant to be both the lines but code review hates me

**Review by harryob — CHANGES_REQUESTED** [review]

File: code/game/machinery/computer/pmc_transfer.dm
Comment (Zonespace27) [reply code/game/machinery/computer/pmc_transfer.dm:95]: Resolved all

**Review by harryob — APPROVED** [review]
code lgtm but i'll likely TM it for some weird edgecases, still requires sprite approval, however it's over 7 days so GA not required

**Comment by harryob** [comment]
and fuck you too merge conflict bot

**Comment by monkeysfist101** [comment]
Waiting on Nanu's input before merge/rejection.

**Comment by Nanu308** [comment]
Hearing with Maintainer team about it as there are multiple opinions if this is fitting or not, and if we want to "encourage" it by adding the mechanics.

**Comment by Nanu308** [comment]
After a lengthy discussion internally we don't approve of this PR's changes and do not wish to encourage the CL into taking Marines out from the USCM and so forth as it is considered desertion etc. and makes little sense IC/lore wise on top of that.

Due to this, the PR will be closed on a design level.

**Comment by Huffie56** [comment]
@Zonespace27 what about you recenter this PR on allowing CL to recruit survivors instead of marines?

**Pull request closed without merging**
